scipy/stats/_stats_py.py

```diff
@@ -294,12 +291,6 @@ def gmean(a, axis=0, dtype=None, weights=None):
     numpy.average : Weighted average
     hmean : Harmonic mean
 
-    Notes
-    -----
-    The geometric average is computed over a single dimension of the input
-    array, axis=0 by default, or all values in the array if axis=None.
-    float64 intermediate and return values are used for integer inputs.
```

Comment:
No: even before we started working on `gmean`, `float64` was not used in intermediate steps with integer input.  

Even with integer input, `float64` is not always the output value. Sure, `np.log` will return a float given integer input, but the precision depends on the precision of the integer input.